mitmproxy/proxy/mode_specs.py

```diff
@@ -209,6 +209,52 @@ def __post_init__(self) -> None:
         _check_empty(self.data)
 
 
+class WireGuardMode(ProxyMode):
+    default_port = 51820
+    transport_protocol: ClassVar[Literal["tcp", "udp"]] = "udp"
+
+    wireguard_conf: str | None = None
+    wireguard_peers: int | None = None
+
+    # noinspection PyDataclass
+    def __post_init__(self) -> None:
+        # settings are empty or a literal "default":
+        if self.data in ["", "default"]:
+            return
+
+        # settings are comma-separated key-value pairs:
+        settings = self.data.split(",")
+
+        # override defaults with specified settings:
+        for kv in settings:
+            if not kv:
+                # empty string after a trailing comma
+                break
+
+            try:
+                key, value = kv.split("=", 1)
+            except ValueError:
+                raise ValueError(f"invalid wireguard mode (not a key=value pair: {kv!r})")
+
+            if key not in ["name", "listen_port", "peers"]:
```

Comment:
What is name here? The configuration file path?

Reply:
It's the base name / prefix for the names of the configuration files that will be generated. 

For example, if `name = "foo", peers = 2`, then the configuration files will be named `foo.conf`, `foo_peer0.conf`, and `foo_peer1.conf`.

I'm sometimes bad at naming things, so I'm open to suggestions for a better name for this option.

Reply:
From a practical perspective, maybe we don't even need the `peers` parameter? If you want multiple peers, you could just start additional instances on different ports and that would work just as well? Then specs could literally just be `wireguard:path/to/wireguard.conf` or simply `wireguard` (for `~/.mitmproxy/wireguard.conf`), avoiding any key/value parsing. :)

Reply:
Would that launch multiple processes? I'm not sure if `boringtun` or `smoltcp` have any global variables that might interfere with running multiple instances in one process.

And you'd still need to generate *at least one* configuration for your peer(s) in some easily accessible location, because otherwise you can't configure your client device. Just writing it to `$(cwd)` seemed to be the simplest option ...

Reply:
No, that's all in the same process. I'd expect them to be free of global state though.

Yes, I think the mechanism here should just be to first determine the filename (a custom passed path or - if none is passed - the default location in ~/. mitmproxy), and then create such a config file with exactly one peer if it does not exist yet. The configuration should then be included in ServerInstance.to_json() so that it can be displayed in mitmweb. :)

Reply:
That won't work though. If we keep only the server configuration, then there's no way to recover the private key for client configurations.

Reply:
One way around this would be to write out a `.key` file next to the `.conf` file. Lack of such a file would not be a critical failure but only stops us from displaying full configuration details in mitmproxy.